markets/perps-market/contracts/modules/PerpsAccountModule.sol

```diff
@@ -100,6 +100,8 @@ contract PerpsAccountModule is IPerpsAccountModule {
         FeatureFlag.ensureAccessToFeature(Flags.PERPS_SYSTEM);
         Account.exists(accountId);
 
+        AsyncOrder.checkPendingOrder(accountId);
```

Comment:
❓ Question:

I see in the `bfp-market` [implementation](https://github.com/Synthetixio/synthetix-v3/blob/75975f9f8ef04a7d44186abffd8a096ddca0d0b8/markets/bfp-market/contracts/modules/MarginModule.sol#L494) of `payDebt()` the strategy for checking order existence solely relies on size and does not concern itself with the order's expiration status. Why is this?

⛳️ Unrelated to this commit:

@davidvuong @0xjocke I noticed when reading the `bfp-market` version of `payDebt()`:

```
PerpMarket.Data storage market = PerpMarket.exists(marketId);
Margin.Data storage accountMargin = Margin.load(accountId, marketId);
Order.Data storage order = market.orders[accountId];
if (order.sizeDelta != 0) {
    revert ErrorUtil.OrderFound();
}
```

loading account margin (`Margin.load(accountId, marketId);`) can be safely moved _after_ the order size conditional logic. This will reduce gas lost if the call throws with `OrderFound`